### ISSUE TITLE
Miscelaneous travis test failiures

### ISSUE THREAD

**Issue by pelson** [matplotlib/matplotlib#2046]
This issue is a bucket for test failures which have occurred on the travis tests, probably related to the fact that the tests are run in parallel.

I'm not unduly worried by these failures - it is something I'd like to fix, but not something which needs to hold up a release.

The problems will inevitably prove hard to reproduce and will be dependent on test execution order (which changes with hash ordering and performance of particular nodes).

**Comment by timj** [comment]
@mdboom I don't think the `cbook.makedirs` code catches all the problems with multi-threading. We are seeing a very rare failure in the LSST test suite when multiple tests are using matplotlib in parallel. I think it is because the try/catch in `makedirs` is assuming that only the last part of the path could have the race and only does a single catch around the entire for loop. If both processes are racing to create path `A/B/C` and both are notice that `B` is missing then the second one to attempt to create `B` will abort when it finds that `B` now exists. It should just continue on with trying to make `C`. I think the try/catch should be in the inner part of the loop around the `os.makedirs` call.

This is the error we get:

```python
import matplotlib.colorbar
  File "/home/vagrant/stack/Linux64/anaconda/2.2.0/lib/python2.7/site-packages/matplotlib/colorbar.py", line 36, in <module>
    import matplotlib.contour as contour
  File "/home/vagrant/stack/Linux64/anaconda/2.2.0/lib/python2.7/site-packages/matplotlib/contour.py", line 27, in <module>
    import matplotlib.texmanager as texmanager
  File "/home/vagrant/stack/Linux64/anaconda/2.2.0/lib/python2.7/site-packages/matplotlib/texmanager.py", line 89, in <module>
    class TexManager:
  File "/home/vagrant/stack/Linux64/anaconda/2.2.0/lib/python2.7/site-packages/matplotlib/texmanager.py", line 124, in TexManager
    mkdirs(texcache)
  File "/home/vagrant/stack/Linux64/anaconda/2.2.0/lib/python2.7/site-packages/matplotlib/cbook.py", line 1003, in mkdirs
    os.makedirs(thispart, mode)
  File "/home/vagrant/stack/Linux64/anaconda/2.2.0/lib/python2.7/os.py", line 157, in makedirs
    mkdir(name, mode)
OSError: [Errno 17] File exists: '/tmp/matplotlib-vagrant'
```
with v1.4.3.

I can file a new issue if you prefer. I just used this as it was open already and related.

**Comment by tacaswell** [comment]
@timj Can you make a new issue?  That makes it easier to track.

I am going to close this issue as until last month it had not been touched for 2 years.

**Comment by tacaswell** [comment]
@timj See https://github.com/matplotlib/matplotlib/pull/5224
